ngraph_bridge/enable_variable_ops/ngraph_replace_variable_modifiers.cc

```diff
@@ -173,6 +200,142 @@ Status ReplaceModifiers(Graph* graph, int graph_id) {
 
       NGRAPH_VLOG(1) << "Replaced ApplyGradientDescent";
     }  // Apply Gradient Descent
+    else if (node->type_string() == "NGraphApplyMomentum") {
+      std::vector<const Edge*> input_edges;
+      TF_RETURN_IF_ERROR(node->input_edges(&input_edges));
+
+      NGRAPH_VLOG(1) << "No of input edges to ApplyMomentum "
+                     << input_edges.size();
+
+      auto input_var = NodeBuilder::NodeOut(input_edges[0]->src(),
+                                            input_edges[0]->src_output());
+      auto input_accum = NodeBuilder::NodeOut(input_edges[1]->src(),
+                                              input_edges[1]->src_output());
+      auto input_lr = NodeBuilder::NodeOut(input_edges[2]->src(),
+                                           input_edges[2]->src_output());
+      auto input_grad = NodeBuilder::NodeOut(input_edges[3]->src(),
+                                             input_edges[3]->src_output());
+      auto input_momentum = NodeBuilder::NodeOut(input_edges[4]->src(),
+                                                 input_edges[4]->src_output());
+      DataType dtype;
+      TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "T", &dtype));
+
+      string new_name_mul = node->name() + "_Mul";
+      NodeBuilder::NodeOut ndef_mul_op;
+      TF_RETURN_IF_ERROR(CreateBinaryOpNode(&ndef_mul_op, new_name_mul, "Mul",
+                                            input_accum, input_momentum, dtype,
+                                            node, graph, true, false));
```

Comment:
This call is the only call to `CreateBinaryOpNode` that has `replace_input`==true`. `replace_output` is false for all.

I haven't looked at the purpose of the 2 booleans, but please make sure they are set to true/false correctly

If `replace_output` is false everywhere, can we remove that argument?

Reply:
No actually, I missed at one place to set it true. Will update, I don't understand how tests are passing.